src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -584,11 +587,12 @@ void ControllerScriptInterfaceLegacy::softTakeover(
         const QString& group, const QString& name, bool set) {
     ControlObject* pControl = ControlObject::getControl(
             ConfigKey(group, name), ControlFlag::AllowMissingOrInvalid);
-    if (!pControl) {
-        return;
-    }
     if (set) {
-        m_st.enable(pControl);
+        auto* pControlPotmeter = qobject_cast<ControlPotmeter*>(pControl);
+        if (!pControlPotmeter) {
+            return;
+        }
+        m_st.enable(gsl::not_null(pControlPotmeter));
```

Comment:
> it exposes SoftTakeover internals to the outside without taking any benefits.

Can you elaborate? Why is the fact that softtakeover only works on `ControlPotMeter`s internal? IMO calling `st.enable(ControlObject*)` and then silently ignoring it because its not a `ControlPotMeter` is a much worse pattern...

>  At least disable() and enable() should take the same parameters.

so both `not_null<ControlPotMeter>`? I explicitly decided against that because it would've forced a null check and a cast onto the caller, both of which are unnecessary.

Reply:
This function here is also invisible when setting the SoftTakeover flag. Currently enable and disable are only called here, that's why I think the conversion to `not_null<ControlPotMeter>` should be done above of set. 
... Or pass `set` es parameter and dispose `disable()`

Reply:
I'm afraid I can't follow you... 

> This function here is also invisible when setting the SoftTakeover flag

What do you mean with invisible? Invisible to whom?

> Currently enable and disable are only called here, that's why I think the conversion to not_null<ControlPotMeter> should be done above of set.

But that would incur the null check and the `qobject_cast` in the unnecessary disable case...

Reply:
Thinking about this more: if we would remove the `ControlObject*` completely from `SoftTakeover` and use `not_null<ControlObject*>` instead, that would allow `SoftTakeover` in a more type-safe fashion, but then we had the cast and the null check everywhere. I explicitly avoided that because I thought you might oppose it, but if you're in favor of that, I'm in favor too.

Reply:
> But that would incur the null check and the qobject_cast in disable case even though its not needed...

Oh yes, that's not optimal. So we may keep it as it is in this PR.  

This started, because I was wondering why the ControlObject type check is now outside of the SoftTakeoverCtrl. Every mapping can have the softtakover flag set an it really dos not matter at which level in the call stack it is rejected, because it is invisible to the mapping author anyway. 

Original the pattern was "here is a CO with the flag set, please do soft takeover as good as you can". 
Now the caller needs to verify first if the the flag is set for the right CO type. We may have an issue if we create another type with Softtakover not inherited by ControlPotmeter.

Reply:
> Every mapping can have the softtakover flag set an it really dos not matter at which level in the call stack it is rejected, because it is invisible to the mapping author anyway.

Yes. Essentially because the ControlPotmeter concept doesn't really exist outside the C++ domain, in JS its just a collection of autogenerated COs.

> We may have an issue if we create another type with Softtakover not inherited by ControlPotmeter.

Yes, but at least now that is obvious since the API explicitly encodes that it is only expecting ControlPotmeters. So the more strongly-typed API is an improvement here IMO.

Reply:
Ok, than keep it like this.